### PR TITLE
Add integration tests for mesh and virtualnode

### PR DESCRIPTION
**Pull request by fawadkhaliq**
**Description of changes**
Added framework/setup for integration tests and initial base integration tests for mesh and virtual node resources. The integration tests will allow testing all things in the controller reconciliation logic, webhooks, injector and state in AWS App Mesh / AWS Cloud Map

- Added integration tests for mesh and virtual node 
- Updated the framework so it can be shared and works for both e2e and integration tests 
- Changed `buildSDKMeshSpec` -> `BuildSDKMeshSpec` to it can be used in the test framework `MeshManager`

**To run tests manually (will be integrated into CI later)**
```
ginkgo -v -r test/integration -- --cluster-kubeconfig=<kubeconfig> --cluster-name=<cluster_name> --aws-region=<aws_region> --aws-vpc-id=<cluster_vpc_id>
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/integration/mesh/mesh_test.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+	appmesh "github.com/aws/aws-app-mesh-controller-for-k8s/apis/appmesh/v1beta2"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/framework"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/framework/utils"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/integration/mesh"
+
+	"github.com/aws/aws-sdk-go/aws"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("Mesh", func() {
```
Comment (achevuru) [inline test/integration/mesh/mesh_test.go:19]: Can we add a testcase for Mesh Delete scenario to validate the use case where we can't delete a mesh if there are appmesh resources under it? Also, to validate that a mesh delete will not automatically trigger delete of appmesh resources under it.Once we delete those resources mesh should be deleted.

-> Create a Mesh
-> Create a VN for ex
-> Delete Mesh -> should be stuck in Terminating state
-> Delete VN and check if Mesh is deleted as well

The above steps will help validate the mesh finalizer code flow..

File: test/integration/virtualnode/virtualnode_test.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+			})
+		}
+		if f.Options.InjectorImage != "" {
+			By("Reset cluster with default injector", func() {
+				f.HelmManager.ResetAppMeshInjector()
+			})
+		}
+	})
+
+	Context("Virtual Node create scenaries", func() {
```
Comment (achevuru) [inline test/integration/virtualnode/virtualnode_test.go:49]: Plan to add Update/Delete scenarios via a new PR? Good to validate these https://github.com/aws/aws-app-mesh-controller-for-k8s/blob/master/webhooks/appmesh/virtualnode_validator.go#L49

File: test/integration/mesh/mesh_test.go (hunk 1 above)
Comment (fawadkhaliq) [reply test/integration/mesh/mesh_test.go:19]: Thanks for the review. All valid scenarios. Yes, I plan to add these scenarios in a follow up. This is just the baseline to get started on the integration tests

File: test/integration/virtualnode/virtualnode_test.go (hunk 2 above)
Comment (fawadkhaliq) [reply test/integration/virtualnode/virtualnode_test.go:49]: ThYes, I plan to add these scenarios in a follow up. This is just the baseline to get started on the integration tests

File: test/integration/virtualnode/virtualnode_test.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+		vnBuilder := &manifest.VNBuilder{
+			ServiceDiscoveryType: manifest.DNSServiceDiscovery,
+		}
+
+		AfterEach(func() {
+			vnTest.Cleanup(ctx, f)
+			meshTest.Cleanup(ctx, f)
+		})
+
+		It("should create a virtual node in AWS", func() {
```
Comment (achevuru) [inline test/integration/virtualnode/virtualnode_test.go:70]: For create scenarios, maybe we can have one DNS and one cloudMap VirtualNode..

File: test/integration/virtualnode/virtualnode_test.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+			backends := []types.NamespacedName{}
+			vn := vnBuilder.BuildVirtualNode(vnName, backends, listeners)
+
+			By("Creating a virtual node resource in k8s", func() {
+				err := vnTest.Create(ctx, f, vn)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			By("validating the virtual node in AWS", func() {
+				err := vnTest.CheckInAWS(ctx, f, mesh, vn)
```
Comment (achevuru) [inline test/integration/virtualnode/virtualnode_test.go:124]: This function needs to be enhanced to check if cloudMap resources are created as well when SD is set to cloudmap. Currently only verifies appmesh resources. Maybe one of us can do this in a followup PR

File: test/integration/virtualnode/virtualnode_test.go (hunk 3 above)
Comment (fawadkhaliq) [reply test/integration/virtualnode/virtualnode_test.go:70]: +1, will be adding all those

**Pull request merged**